src/modules/Tab/Tab.js

```diff
@@ -88,31 +92,52 @@ class Tab extends Component {
   }
 
   renderItems() {
-    const { panes, renderActiveOnly } = this.props
+    const { panes, renderActiveOnly, children } = this.props
     const { activeIndex } = this.state
 
-    if (renderActiveOnly) return _.invoke(_.get(panes, `[${activeIndex}]`), 'render', this.props)
-    return _.map(panes, ({ pane }, index) =>
-      TabPane.create(pane, {
+    const isMissingChildren = childrenUtils.isNil(children)
+
+    const items = isMissingChildren ? panes : children
+
+    if (renderActiveOnly) {
+      if (!isMissingChildren) {
+        return _.get(items, `[${activeIndex}]`)
+      }
+      return _.invoke(_.get(items, `[${activeIndex}]`), 'render', this.props)
+    }
+
+    return _.map(items, (item, index) => {
+      let pane = item
+
+      if (isMissingChildren) {
+        pane = item.pane || item.render(this.props)
+      }
+
+      return TabPane.create(pane, {
         overrideProps: {
           active: index === activeIndex,
         },
-      }),
-    )
+      })
+    })
   }
 
   renderMenu() {
-    const { menu, panes, menuPosition } = this.props
+    const { menu, panes, menuPosition, children } = this.props
+
     const { activeIndex } = this.state
 
     if (menu.tabular === true && menuPosition === 'right') {
       menu.tabular = 'right'
     }
 
+    const items = childrenUtils.isNil(children)
+      ? _.map(panes, 'menuItem')
+      : _.map(children, child => child.props.menuItem)
```

Comment:
Sorry, but this change cannot be merged because it violates our API.

The parent component should never know anything about the child component. This is one of reasons for the shorthand API existance.

Reply:
I understand. Can I take another crack at this where the `menuItem` isn't pulled from the child? Something like this perhaps:
```
<Tab menu={['Tab 1', 'Tab 2']}>
    <Tab.Pane>Tab 1 Content</Tab.Pane>
    <Tab.Pane>Tab 2 Content</Tab.Pane>
</Tab>
```
Where `menu` is a string, node, or menu shorthand.

Reply:
I'm not sure that you're actually need this.  These can be done with `Menu` component.

```jsx
    <Menu attached tabular>
      <Menu.Item active>Tab 1</Menu.Item>
      <Menu.Item>Tab 2</Menu.Item>
      <Menu.Item>Tab 3</Menu.Item>
    </Menu>

    <Tab.Pane active>Tab 1 Content</Tab.Pane>
    <Tab.Pane active={false}>Tab 2 Content</Tab.Pane>
```

https://codesandbox.io/s/66z9j2l3k

Reply:
I see what you are saying, but your example would require the user to implement the state that manages the active panes and such. My goal was to not have the user have to worry about that and still have `Tab.Pane` be a valid child of `Tab`, similar to how other components work.